weather_stats/weatherstatsapp/weather/src/Search.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from "react"
+import Temperature from "./Temperature"
+import SearchForm from "./SearchForm"
+import ResultListItemBoundary from "./ResultListItemBoundary"
+import ResultListItemPlace from "./ResultListItemPlace"
+
+const Search = () => {
+    const [searchInput, setSearchInput] = useState("")
+    const [resultList, setResultList] = useState([])
+    const [weather, setWeather] = useState(null)
+    const [selectedCity, setSelectedCity] = useState(null)
+    const [loading, setLoading] = useState(false)
+    const [weatherLoading, setWeatherLoading] = useState(false)
+    const [error, setError] = useState(false)
+
+    const handleResultClick = async (clickedData) => {
+        setWeatherLoading(true)
+        setWeather(true)
+        setError(false)
+        await new Promise((resolve) => setTimeout(resolve, 600))
+        fetch(`/api/v1/weather?lat=${clickedData.lat}&long=${clickedData.lon}`)
```

Comment:
(Style) It's really confusing that there are pieces when `async/await` is used, and then we have the switch to the Promises API. Lets be consistent and use `async/await` here. This means that a `try/catch/finally` is going to be necessary. You can update flags in the `finally` block.